README.md

```diff
@@ -454,63 +447,53 @@ matches-ignore:
+At startup, ggshield attempts to load environment variables from different environment files in the following order:
 
 - path pointed to by the environment variable `GITGUARDIAN_DOTENV_PATH`
 - `.env` at your current work directory.
 - `.env` at the root of the current git directory
 
 Only one file will be loaded of the three.
 
-Reference of current Environment Variables that affect `ggshield`:
+Reference of current environment variables supported by ggshield:
 
-```yaml
-GITGUARDIAN_API_KEY: [Required] API Key for the GitGuardian API.
+- `GITGUARDIAN_API_KEY`: API Key for the GitGuardian API. Use this if you don't want to use the `ggshield auth` commands.
 
-GITGUARDIAN_URL: Custom URL of the GitGuardian dashboard. The API URL will be inferred from it.
+- `GITGUARDIAN_INSTANCE`: Custom URL of the GitGuardian dashboard. The API URL will be inferred from it.
 
-GITGUARDIAN_API_URL: (Deprecated use GITGUARDIAN_URL instead) Custom URL for the scanning API.
+- `GITGUARDIAN_API_URL`: Custom URL for the scanning API. Deprecated, use `GITGUARDIAN_INSTANCE` instead.
 
-GITGUARDIAN_DONT_LOAD_ENV: If set to any value environment variables won't be loaded from a file.
+- `GITGUARDIAN_DONT_LOAD_ENV`: If set to any value then environment variables won't be loaded from a file.
 
-GITGUARDIAN_DOTENV_PATH: If set to a path, `ggshield` will attempt to load the environment from the specified file.
+- `GITGUARDIAN_DOTENV_PATH`: If set to a path, ggshield will attempt to load the environment from the specified file.
 
-GITGUARDIAN_TIMEOUT: If set to a float, `ggshield secret scan pre-receive` will timeout
-after the specified value. Set to 0 to disable the timeout
+- `GITGUARDIAN_TIMEOUT`: If set to a float, `ggshield secret scan pre-receive` will timeout after the specified value. Set to 0 to disable the timeout.
 
-GITGUARDIAN_MAX_COMMITS_FOR_HOOK: if set to an int, `ggshield secret scan pre-receive` and `ggshield secret scan pre-push`
-will not scan more than the specified value of commits in a single scan.
+- `GITGUARDIAN_MAX_COMMITS_FOR_HOOK`: If set to an int, `ggshield secret scan pre-receive` and `ggshield secret scan pre-push` will not scan more than the specified value of commits in a single scan.
 
-GITGUARDIAN_CRASH_LOG: If set to True, ggshield will display a full traceback
-when crashing
-```
+- `GITGUARDIAN_CRASH_LOG`: If set to True, ggshield will display a full traceback when crashing.
 
 ## On-premises configuration
 
-GitGuardian shield can be configured to run on your on-premises dashboard, request an API key from your dashboard administrator.
+ggshield can be configured to run on your on-premises dashboard.
 
-You can modify your environment variables to include:
+First, you need to point ggshield to your instance, by either defining the `instance` key in your `.gitguardian.yaml` configuration file or by defining the `GITGUARDIAN_INSTANCE` environment variable.
```

Comment:
Is it a necessary step? If yes, it needs to be also documented in public doc.

Isn't doing the authentication with `--instance https://myinstance` enough?

Reply:
Yes, this is required: the value of `auth login --instance` option is used to decide the instance to authenticate against and is stored with the associated PAT, but it does not define the repository instance.

Going to add this to the public doc MR.